src/generated/resources/schema.graphql

```diff
@@ -59,6 +59,27 @@ type ConfluentLocalKafkaCluster {
   uri: String!
 }
 
+type DirectConnection {
+  id: String!
+  kafkaCluster: DirectKafkaCluster
+  name: String!
+  schemaRegistry: DirectSchemaRegistry
+  type: ConnectionType!
+}
+
+type DirectKafkaCluster {
+  bootstrapServers: String!
+  connectionId: String!
+  id: String!
+  uri: String
```

Comment:
Out of curiosity, what does `uri` hold in the case of a `DirectKafkaCluster`?

Reply:
Good question. First, in the GraphQL schema, the `Cluster` interface defines `uri` and `DirectKafkaCluster` inherits the field whether or not it needs it.

Second, I think that's okay, because the `CCloudKafkaCluster` and `ConfluentLocalKafkaCluster` subtypes expose the Kafka REST proxy URL via that `uri` field, and we will want the `DirectKafkaCluster` to be able to expose the Kafka REST proxy URL if there is one and we know about it. I do anticipate in the near future enhancing the `kafka_cluster` config in direction connections to optionally include the REST endpoint, either purely as informational, or to be able to specify the REST endpoint instead of `bootstrapServers` so we can then look up the latter (and more Kafka cluster metadata) from the former like we currently do in the local connections.

Bottom line: it is currently a bit weird now, but I think we'll make it work in the next sprint.

Reply:
Thanks for the clarification! I didn't know we want to support interacting with `DirectKafkaCluster`s via the REST proxy. Also, as you said, the [KafkaClusterConfig doesn't seem to offer the uri field](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/java/io/confluent/idesidecar/restapi/models/ConnectionSpec.java#L246) right now so I'm wondering how/if we'd fill the `uri` field of the GraphQL API's `DirectKafkaCluster`. Let's get back to it in one of the next sprints per your suggestion.

Reply:
> I didn't know we want to support interacting with DirectKafkaClusters via the REST proxy.
Just for bootstrapping to collect metadata like cluster ID. We still plan to use the internal REST proxy for functionality rather than rely upon the provided REST proxy.